app/src/main/java/org/gdg/frisbee/android/activity/MainActivity.java

```diff
@@ -203,31 +190,39 @@ protected void onResume() {
         }
     }
 
-    private void selectChapter(Chapter chapter) {
-        mViewPagerAdapter.setSelectedChapter(chapter);
-        getSupportActionBar().setSelectedNavigationItem(mSpinnerAdapter.getPosition(chapter));
-    }
-
-    private void initChapters(ArrayList<Chapter> chapters) {
+    /**
+     * Initializes ViewPager, SlidingTabLayout, and Spinner Navigation. 
+     * 
+     * If the selectedChapter is null, tries to get it from Intent Extras,
+     *                                  if not found, gets the first Chapter.
+     * This function is called after cache query or network call is success,
+     * or after an orientation change using savedInstance.
+     *  
+     * @param chapters Chapter array to be initialized, never null.
+     * @param selectedChapter selectedChapter, can be null.
+     */
+    private void initChapters(@NonNull ArrayList<Chapter> chapters,

```

Comment:
Please change this method into two methods. If you are doing code review and you read the following `initChapters(chapters, null);` and before it was just `initChapters(chapters);` you will wonder what that `null` argument is and you have to check the new signature.

It's better to have the method just do one thing instead of two (one for null, one for not null) so that it's obvious what's happening just by reading the line.